content/learning-paths/cross-platform/integer-vs-floats/integer-float-conversions.md

```diff
@@ -0,0 +1,263 @@
+|            int64_t   |             |             |             |    int128_t*  |
+|            fp16**    |             |    float    |   double    |  long double* |
+|            bf16**    |             |    float    |   double    |  long double* |
+|            float     |             |             |   double    |  long double* |
+|            double    |             |             |             |  long double* |
+
+(`*`) depends on the actual compiler support
+(`**`) depends on hardware support
+
+Similar promotions take place for unsigned integers.
+
+### Demotions
+
+Respectively, if `A` is `float` and `B` is `double` but `C` is `float` then we have a demotion.
+This is actually a very risky conversion, as bits are lost and it is discouraged and should be avoided unless in specific circumstances when you know exactly what you are doing.
+Unfortunately, this is where the programming language matters. In some cases, C++ will catch such a demotion (called a *narrowing conversion* in C++) and will issue a relevant warning,
+C does not allow for such a warning. The C compiler will not issue a warning and it's easy for a conversion bug to creep in your code, and sometimes these can be very hard to detect bugs.
+Even on C++ there is a catch however, the compiler will only issue such a warning when using bracket initialization, not assignment between values. You will have a chance to see this in detail in the next section.
+
+Here is a similar list for possible demotions:
+
+| Data type (stdint.h)  | Demoted to                                             |
+| -------------------- | ----------- | ----------- | ----------- | ------------- |
+|            int16_t   |   int8_t    |             |             |               |
+|            int32_t   |   int8_t    |   int16_t   |             |               |
+|            int64_t   |   int8_t    |   int16_t   |   int32_t   |               |
+|            fp16**    |             |    float    |   double    |  long double* |
+|            bf16**    |             |    float    |   double    |  long double* |
+|            float     |             |             |   double    |  long double* |
+|            double    |             |             |             |  long double* |
+
+Again unsigned integers are demoted to similar types.
+
+(`*`) depends on the actual compiler support
+(`**`) depends on hardware support
+
+### Type conversions
+
+Some might argue that conversion of an `int16_t` to `float` or `double` is a promotion, but it's not as simple as that. While a demotion does not need any instruction to take place, usually a promotion requires an instruction to zero or sign-extend the contents of a register. However this can be achieved using other ways also, when the compiler can detect a specific pattern it can skip those zero/sign-extend instructions and solve the problem by mere shuffling/rearrangements on the bytes. For example, this example demonstrates it:
```

Comment:
I'd change
> demotion does not need

to

> demotion does not *always* need

other than that, looks good, thanks!

Reply:
fixed.